src/hosted-buttons/index.js

```diff
@@ -0,0 +1,71 @@
+/* @flow */
+
+import { getMerchantID } from "@paypal/sdk-client/src";
+
+import { getButtonsComponent } from "../zoid/buttons";
+
+import {
+  buildHostedButtonCreateOrder,
+  buildHostedButtonOnApprove,
+  getHostedButtonDetails,
+  renderForm,
+} from "./utils";
+import type {
+  HostedButtonsComponent,
+  HostedButtonsComponentProps,
+  HostedButtonsInstance,
+} from "./types";
+
+export const getHostedButtonsComponent = (): HostedButtonsComponent => {
+  function HostedButtons({
+    hostedButtonId,
+  }: HostedButtonsComponentProps): HostedButtonsInstance {
+    const Buttons = getButtonsComponent();
+    const render = (selector) => {
+      // The SDK supports mutiple merchant IDs, but hosted buttons only
+      // have one merchant id as a query parameter to the SDK script.
+      // https://github.com/paypal/paypal-sdk-client/blob/c58e35f8f7adbab76523eb25b9c10543449d2d29/src/script.js#L144
+      const merchantId = getMerchantID()[0];
+
+      getHostedButtonDetails({ hostedButtonId }).then(
+        ({ html, htmlScript, style }) => {
+          renderForm({ html, htmlScript, selector });
+
+          // $FlowFixMe
+          Buttons({
+            style,
+            hostedButtonId,
+            onInit(data, actions) {
+              // disable the button, listen for input changes,
+              // and enable the button when the form is valid
+              // using actions.disable() and actions.enable()
+              window[`__pp_form_fields_${hostedButtonId}`]?.onInit?.(
```

Comment:
could we return callbacks from `renderForm` and use them here instead of trying to find it on the window? does that make sense?

```
const formComponent = renderForm(...)
formComponent.onInit
```

Reply:
oo I like this idea - a254ae90

but I think they have to be called on `window` (or some other global on `window`) because the method is loaded dynamically after fetching the hosted button details.

Reply:
we chatted offline about this, going to keep this as is